pkg/asset/machines/aws/awsmachines.go

```diff
@@ -160,3 +165,62 @@ func CapaTagsFromUserTags(clusterID string, usertags map[string]string) (capa.Ta
 	}
 	return tags, nil
 }
+
+// CapaIgnitionWithCertBundleAndProxy generates CAPA ignition config with cert and proxy information.
+func CapaIgnitionWithCertBundleAndProxy(userCA string, proxy *types.Proxy) (*capa.Ignition, error) {
+	carefs, err := parseCertificateBundle([]byte(userCA))
+	if err != nil {
+		return nil, err
+	}
+	return &capa.Ignition{
+		Version: "3.2",
+		TLS: &capa.IgnitionTLS{
+			CASources: carefs,
+		},
+		Proxy: capaIgnitionProxy(proxy),
+	}, nil
+}
+
+// TODO: try to share this code with ignition.bootstrap package?
+// parseCertificateBundle loads each certificate in the bundle to the CAPA
+// carrier type, ignoring any invisible character before, after and in between
+// certificates.
+func parseCertificateBundle(userCA []byte) ([]capa.IgnitionCASource, error) {
+	userCA = bytes.TrimSpace(userCA)
+
+	var carefs []capa.IgnitionCASource
+	for len(userCA) > 0 {
+		var block *pem.Block
+		block, userCA = pem.Decode(userCA)
+		if block == nil {
+			return nil, fmt.Errorf("unable to parse certificate, please check the certificates")
+		}
+
+		carefs = append(carefs, capa.IgnitionCASource(dataurl.EncodeBytes(pem.EncodeToMemory(block))))
```

Comment:
Hm, I suspect we don't want `dataurl.EncodeBytes` here, but I do see the e2e proxy test successfully installed.  Do you have any more background on this (no worries if you don't, just checking if I'm missing something)? I would think `dataurl` would be used when passing in ignition, but I wouldn't think that would be needed for the CAs...

Reply:
As discussed on Slack, it's required as per spec: https://coreos.github.io/ignition/configuration-v3_2/